scarlet2/initialization.py

```diff
@@ -0,0 +1,304 @@
+            perimeter_values.extend(array_2d[i][-1] for i in range(1, rows - 1))
+
+    return perimeter_values
+
+
+# create a 2D Gaussian for simple feature free Initializations
+def gaussian_2d(x, y, mu_x, mu_y, sigma_x, sigma_y):
+    """2D Gaussian function.
+    Parameters
+    ----------
+    x : jnp.ndarray
+    y : jnp.ndarray
+    mu_x : float mean of the Gaussian in the x direction
+    mu_y : float mean of the Gaussian in the y direction
+    sigma_x : float standard deviation of the Gaussian in the x direction
+    sigma_y : float standard deviation of the Gaussian in the y direction
+
+    Returns
+    -------
+    jnp.ndarray 2D Gaussian profile
+    """
+    return jnp.exp(
+        -((x - mu_x) ** 2 / (2 * sigma_x**2) + (y - mu_y) ** 2 / (2 * sigma_y**2))
+    )
+
+
+@partial(jax.jit, static_argnums=0)
+def create_gaussian_array(boxsize, sigma_x=1, sigma_y=1, theta=0):
+    """Create a 2D array with a Gaussian profile in the center.
+    Parameters
+    ----------
+    boxsize : int size of the bounding box around source
+    sigma_x : float standard deviation of the Gaussian in the x direction
+    sigma_y : float standard deviation of the Gaussian in the y direction
+    theta : float rotation angle of the 2D Gaussian
+
+    Returns
+    -------
+    jnp.ndarray 2D array with a Gaussian profile rotated by theta
+    """
+    x = jnp.linspace(-boxsize / 2, boxsize / 2, boxsize)
+    x, y = jnp.meshgrid(x, x)
+    # Center of the Gaussian profile
+    mu_x, mu_y = 0, 0
+    # Generate the Gaussian profile
+    gaussian_profile = gaussian_2d(x, y, mu_x, mu_y, sigma_x, sigma_y)
+    gaussian_profile = (gaussian_profile - jnp.min(gaussian_profile)) / (
+        jnp.max(gaussian_profile) - jnp.min(gaussian_profile)
+    )
+    gaussian_profile = jnp.array(gaussian_profile, dtype=jnp.float32)
+    gaussian_profile = jnp.expand_dims(gaussian_profile, axis=-1)
+    data = gaussian_profile #dm_pix.rotate(gaussian_profile, theta)
+    return jnp.squeeze(data)
+
+def init_simple_morph(
+    observation, center, psf_sigma=0.5, noise_thresh=20, corr_thresh=0.8
+):
+    """
+    Initialize the morphology of a source by fitting a 2D Gaussian to the cutout of the source.
+    The boxsize will initially be fit as a compact point-source and then expanded until the snr
+    is below a desired threshold.
+
+    Parameters
+    ----------
+    observation: `~scarlet.Observation`
+    center: tuple --> center of source
+    psf_sigma: float --> initial psf used in model
+    noise_thresh: float --> threshold for noise
+    corr_thresh: float --> threshold for spectrum correlations
+
+    Returns
+    -------
+    morph: `jnp.array` --> 2D Gaussian morphology
+    """
+    sigma = psf_sigma
+    converged = False
+    peak_spec = init_spectrum(observation, center, correct_psf=True)
+    spectra_prev = peak_spec
+    noise_rms = 1 / np.sqrt(ma.masked_equal(observation.weights, 0))
+    ma.set_fill_value(noise_rms, np.inf)
```

Comment:
This breaks in the case where `observation.weights=None`. I think we should either document  that we should note use None weights, or take into account this case in the code

Reply:
For now added assertion that weights are not none